allennlp/training/trainer.py

```diff
@@ -325,12 +342,16 @@ def _train_epoch(self, epoch: int) -> Dict[str, float]:
 
             self.optimizer.zero_grad()
 
-            loss = self.batch_loss(batch_group, for_training=True)
-
-            if torch.isnan(loss):
-                raise ValueError("nan loss encountered")
+            batches_for_step = list(lazy_groups_of(iter(batch_group), len(self._cuda_devices)))
```

Comment:
The conversion `list` seems unnecessary.

Reply:
It's needed because I need to know how many there are to scale the loss.

Reply:
👍